### ISSUE TITLE
Clocks.fmu: access common variables outside activateModelPartition3()

### ISSUE THREAD

**Issue by masoud-najafi** [modelica/Reference-FMUs#464]
Hi,
I'm checking the code of Clocks.fmu. I have a bit doubt about the activateModelPartition3 partition. Actually, in activateModelPartition3() shouldn't the following code be protected by lockPreemption/UnlockPreemption?

    M(output3) = 1000;   // this is suposed to find its way into mp2
    M(totalInClockTicks)++;
    // set output clocks
    M(outClock) = ((M(outClock) == false) && (M(totalInClockTicks) % 5 == 0));

If not, any reason for that?
Thanks,
Masoud

**Comment by MBlesken** [comment]
@masoud-najafi you are right, and there are a couple more lines where lock/unlock has to be adjusted.
